### PR TITLE
Enable cursor when href channel present

### PR DESCRIPTION
**Pull request by melissatdiamond**
Fix for issue #3274

### CODE REVIEW

**Comment by melissatdiamond** [comment]
Ready for review

File: site/docs/encoding.md [hunk 1]
```diff
@@ -236,7 +236,7 @@ Similar to [`detail`](#detail), you can use an array of field definitions. Vega-
 {:#href}
 ## Hyperlink Channel
 
-By setting the `href` channel, a mark becomes a hyperlink. The specified URL is loaded upon a mouse click. The [`cursor` mark property](mark.html#hyperlink) can be set to `pointer` to serve as affordance for hyperlinks.
+By setting the `href` channel, a mark becomes a hyperlink. The specified URL is loaded upon a mouse click. When the `href` channel is specified, the [`cursor` mark property](mark.html#hyperlink) is set to `pointer` to serve as affordance for hyperlinks.
```
Comment (kanitw) [inline site/docs/encoding.md]: `pointer` => `"pointer"`

Please also add "by default".

File: site/docs/mark/mark.md [hunk 2]
```diff
@@ -91,7 +91,7 @@ Note: If [mark property encoding channels](encoding.html#mark-prop) are specifie
 {:#hyperlink}
 ### Hyperlink Properties
 
-Marks can act as hyperlinks when the `href` property or [channel](encoding.html#href) is defined. A `cursor` property can also be provided to serve as affordance for the links.
+Marks can act as hyperlinks when the `href` property or [channel](encoding.html#href) is defined. When the `href` property is specified, the [`cursor` mark property](mark.html#hyperlink) is set to `pointer` to serve as affordance for hyperlinks.
```
Comment (kanitw) [inline site/docs/mark/mark.md]: `pointer` => `"pointer"`

Please also add "by default".

File: src/compile/mark/init.ts [hunk 3]
```diff
@@ -30,9 +30,23 @@ export function normalizeMarkDef(mark: Mark | MarkDef, encoding: Encoding<string
...
+  const specifiedCursor = markDef.cursor;
+  if (specifiedCursor === undefined) {
+    markDef.cursor = cursor(markDef, encoding);
+  }
+
   return markDef;
 }
 
+function cursor(markDef: MarkDef, encoding: Encoding<String>) {
+  if (encoding.href || markDef.href) {
```
Comment (kanitw) [inline src/compile/mark/init.ts]: Do we have href as a mark config?  If so, should we consider those too?

Comment (domoritz) [reply src/compile/mark/init.ts]: I didn't think so but I will check.

**Comment by domoritz** [comment]
@kanitw Should we do anything about `baseEncodeEntry`?

**Comment by kanitw** [comment]
Not sure why is that related given cursor is currently *not* a channel?

**Comment by domoritz** [comment]
https://github.com/vega/vega-lite/issues/3274#issuecomment-365707945

**Pull request merged**